src/main/java/org/osc/controller/nsc/restserver/api/InspectionHookApis.java

```diff
@@ -0,0 +1,163 @@
+    private String id;
+
+    public void init(String id) throws Exception {
+        if (id != null) {
+            if (id.equals(this.id)) {
+                return;
+            }
+        } else {
+            return;
+        }
+
+        Properties props = new Properties();
+        props.setProperty(JDBC_URL, "jdbc:h2:./nscPlugin_" + id + ";MVCC\\=TRUE;LOCK_TIMEOUT\\=10000;MV_STORE=FALSE;");
+        props.setProperty(JDBC_USER, "admin");
+        props.setProperty(JDBC_PASSWORD, "admin123");
+        DataSource ds = null;
+        try {
+            ds = this.jdbcFactory.createDataSource(props);
+        } catch (SQLException error) {
+            LOG.error(error);
+            throw new IllegalStateException(error.getMessage(), error);
+        }
+        EntityManager em = this.resourceFactory
+                .getProviderFor(this.builder, singletonMap("javax.persistence.nonJtaDataSource", (Object) ds), null)
+                .getResource(this.txControl);
+        this.em = em;
+        this.sdnApi = new SampleSdnRedirectionApi(this.txControl, this.em);
+        this.utils = new RedirectionApiUtils(this.em, this.txControl);
+        this.id = id;
+
+    }
+
+    @Path("/controller/{id}/inshook")
+    @POST
+    public String createInspectionHook(@PathParam("id") String id, InspectionHookEntity entity) throws Exception {
+
+        LOG.info(String.format("Creating inspection hook with inspection port id %s",
+                entity.getInspectionPort().getElementId()));
+        init(id);
+        return this.sdnApi.installInspectionHook(entity.getInspectedPort(), entity.getInspectionPort(), entity.getTag(),
+                entity.getEncType(), entity.getOrder(), entity.getFailurePolicyType());
+    }
+
+    @Path("/controller/{id}/inshook/{inshookId}")
+    @PUT
+    public InspectionHookEntity updateInspectionHook(@PathParam("id") String id,
+            @PathParam("insportId") String insportId, @PathParam("inshookId") String inshookId,
+            InspectionHookEntity entity) throws Exception {
+        entity.setHookId(inshookId);
+        this.sdnApi.updateInspectionHook(entity);
+        return entity;
+    }
+
+    @Path("/controller/{id}/inshook/{inshookId}")
+    @DELETE
+    public void deleteInspectionHook(@PathParam("id") String id, @PathParam("inshookId") String inshookId)
+            throws Exception {
+        if (inshookId == null) {
+            throw new IllegalArgumentException("Attempt to delete null inspection hook port");
+        }
+        LOG.info(String.format("Deleting the inspection hook element for id %s ", inshookId));
+        init(id);
+        this.sdnApi.removeInspectionHook(inshookId);
+    }
+
+    @Path("/controller/{id}/inshook/")
+    @GET
+    public List<String> getInspectionHookIds(@PathParam("id") String id) throws Exception {
+        LOG.info("Listing inspection hook ids'");
+        init(id);
+
+        return this.txControl.supports(() -> {
```

Comment:
instead of having this txcontrol here you can actually create a method in the `SampleSdnRedirectionApi` and call it here. This should be fine since this class has the type `SampleSdnRedirectionApi` instead of the SDK interface. This way the `map` approach I mentioned above can still work here since you would no longer need the txControl.